src/Weaviate.Client/RolesClient.cs

```diff
@@ -0,0 +1,127 @@
+using Weaviate.Client.Models;
+
+namespace Weaviate.Client;
+
+/// <summary>
+/// Client for managing roles and role permissions.
+/// </summary>
+public class RolesClient
+{
+    private readonly WeaviateClient _client;
+
+    internal RolesClient(WeaviateClient client) => _client = client;
+
+    public async Task<IEnumerable<RoleInfo>> List()
+    {
+        var roles = await _client.RestClient.RolesList();
+        return roles.Select(ToModel);
+    }
+
+    public async Task<RoleInfo?> Get(string id)
+    {
+        var role = await _client.RestClient.RoleGet(id);
+        return role is null ? null : ToModel(role);
+    }
+
+    public Task<bool> Create(string name, IEnumerable<PermissionInfo> permissions)
+    {
+        var dto = new Rest.Dto.Role
+        {
+            Name = name,
+            Permissions = permissions
+                .Select(p => new Rest.Dto.Permission
+                {
+                    Action = p.Action.FromEnumMemberString<Rest.Dto.PermissionAction>(),
+                })
+                .ToList(),
+        };
+        return _client.RestClient.RoleCreate(dto);
+    }
+
+    public Task<bool> Delete(string id) => _client.RestClient.RoleDelete(id);
+
+    public Task<bool> AddPermissions(string id, IEnumerable<PermissionInfo> permissions)
+    {
+        var dtos = permissions.Select(p => new Rest.Dto.Permission
+        {
+            Action = p.Action.FromEnumMemberString<Rest.Dto.PermissionAction>(),
+        });
+        return _client.RestClient.RoleAddPermissions(id, dtos);
+    }
+
+    public Task<bool> RemovePermissions(string id, IEnumerable<PermissionInfo> permissions)
+    {
+        var dtos = permissions.Select(p => new Rest.Dto.Permission
+        {
+            Action = p.Action.FromEnumMemberString<Rest.Dto.PermissionAction>(),
+        });
+        return _client.RestClient.RoleRemovePermissions(id, dtos);
+    }
+
+    public Task<bool?> HasPermission(string id, PermissionInfo permission)
+    {
+        var dto = new Rest.Dto.Permission
+        {
+            Action = permission.Action.FromEnumMemberString<Rest.Dto.PermissionAction>(),
+        };
+        return _client.RestClient.RoleHasPermission(id, dto);
+    }
+
+    public async Task<IEnumerable<UserRoleAssignment>> UserAssignments(string roleId)
+    {
+        var list = await _client.RestClient.RoleUserAssignments(roleId);
+        return list.Select(a => new UserRoleAssignment(a.userId, a.userType.ToEnumMemberString()!));
+    }
+
+    public async Task<IEnumerable<GroupRoleAssignment>> GroupAssignments(string roleId)
```

Comment:
Should we add "GetXXXAssignments" to make things extra clear? Same syntax as in python